hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmCriteriaNodeBuilder.java

```diff
@@ -914,6 +914,9 @@ void checkMultiselect(List<? extends JpaSelection<?>> selections) {
 	@Override
 	public <T> SqmLiteral<T> literal(T value) {
 		if ( value == null ) {
+			if ( jpaComplianceEnabled ) {
+				throw new IllegalArgumentException( "literal value cannot be null" );
+			}
```

Comment:
Shouldn't we maybe throw an error in general since there is a special `nullLiteral` method?

Reply:
I mean what is the real reason to not accept null here?  Aside from the spec saying we should not and the TCK checking it

Reply:
To avoid potential erroneous passing of a null value and due to that producing an untyped null literal. It's no big deal maybe, but I'd prefer to align with the JPA spec here.